tests/unit/s3routes/routesUtils/redirectRequest.spec.js

```diff
@@ -22,32 +23,87 @@ describe('routesUtils.redirectRequest', () => {
         const objectKey = '';
         const redirectLocationHeader = true;
 
-        it('should redirect to absolute url', () => {
+        it('should redirect to absolute url (path style bucket)', () => {
+            const requestMock = new DummyRequest({
+                socket: { encrypted: false },
+                gotBucketNameFromHost: false,
+                bucketName: 'test',
+            });
             const responseMock = new HttpResponseMock();
             const routing = {
                 redirectLocationHeader,
                 protocol: 'https',
-                hostName: 'scality.com/test',
+                hostName: 'scality.com',
             };
             routesUtils.redirectRequest(
-                routing, objectKey, encrypted, responseMock, hostHeader,
+                routing, objectKey, requestMock, responseMock,
                 corsHeaders, new DummyRequestLogger(),
             );
             assert.strictEqual(responseMock.statusCode, 301);
             assert.strictEqual(responseMock._body, null);
             assertCors(responseMock);
+            console.log({ location: responseMock._headers.Location })
             assert.strictEqual(responseMock._headers.Location, 'https://scality.com/test');
         });
 
-        it('should redirect to relative url', () => {
+        it('should redirect to absolute url (dns style bucket)', () => {
```

Comment:
```suggestion
        it('should redirect to absolute url (virtual-hosted style bucket)', () => {
```

https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html